artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -528,8 +543,17 @@ public void decodeHeadersAndProperties(final ByteBuf buffer) {
    private void decodeHeadersAndProperties(final ByteBuf buffer, boolean lazyProperties) {
       messageIDPosition = buffer.readerIndex();
       messageID = buffer.readLong();
-
-      address = SimpleString.readNullableSimpleString(buffer);
+      int b = buffer.readByte();
+      if (b != DataConstants.NULL) {
+         final int length = buffer.readInt();
+         if (keysInterner != null) {
+            address = keysInterner.intern(buffer, length);
```

Comment:
If you are having separate pools of last seen SimpleStrings instead of global interning for typed properties keys and values, you should probably pool address's separate from those. Imagine the pool is size 32 default, very quickly be used up and find typedproperty keys are invalidating address's in that last seen pool

Reply:
This is a very good point..yes, agree!

Reply:
btw, on another note, correlation id will kill you as each one should be unique, so would constantly invalidate, need to be a bit careful on property values to pool